src/plot_api/plot_api.js

```diff
@@ -151,9 +151,12 @@ Plotly.plot = function(gd, data, layout, config) {
         makePlotFramework(gd);
     }
 
-    // save initial axis range once per graph
-    if(graphWasEmpty) Plotly.Axes.saveRangeInitial(gd);
-
+    if(graphWasEmpty) {
+        // save initial axis range once per graph
+        Plotly.Axes.saveRangeInitial(gd);
+        // save initial show spikes once per graph
+        Plotly.Axes.saveShowSpikeInitial(gd);
+    }
```

Comment:
Just a heads up, this is going to cause merge conflicts, as in the new axis constraints feature this block got moved - https://github.com/plotly/plotly.js/pull/1522/files#diff-2941ab69a12080c0633ff4ac8ea3aa83R274